haystack/components/rankers/transformers_similarity.py

```diff
@@ -37,9 +38,10 @@ class TransformersSimilarityRanker:
     def __init__(
         self,
         model_name_or_path: Union[str, Path] = "cross-encoder/ms-marco-MiniLM-L-6-v2",
-        device: str = "cpu",
+        device: Optional[str] = "cpu",
```

Comment:
Should we leave the default as `cpu`, or would it make sense to standardize? For example, the default for the ExtractiveReader is None. What do you think @julian-risch?

Reply:
Standardizing sounds good. No surprises for users are better. Let's see whether `cpu` or `None` is the better default. If we make `cpu` the default then all the pipeline yaml files in applications and all our tutorials need to set it explicitly to `gpu` or `None`. If users forget to do this then their application will be slower but they might not understand why. That's bad. 
I like `None` as a default because we will then fallback to cpu and show a message to the user if torch is not installed. If you change the default to cpu, please also update the docstring.

Reply:
I saw that `SentenceTransformersTextEmbedder` and `SentenceTransformersDocumentEmbedder` use `cpu` if no device is specified in the init. The advantage of having `cpu` as the default is that the behavior is more predictable.